### ISSUE TITLE
Content in dialog cannot be loaded

### ISSUE THREAD

**Issue by ZheSun88** [vaadin/flow-components#5719]
### Description

running Vaadin 24.3.2 and V24.3.3 on bakery, content in new order dialog are not loaded. 

### Expected outcome

content are loaded as in the image

![image](https://github.com/vaadin/flow-components/assets/31067185/95c6bd6d-e748-4fed-92e5-4ad26a2ce8b0)

### Minimal reproducible example

- git clone git@github.com:vaadin/bakery-app-starter-flow-spring.git
- mvn spring-boot:run -Pit,production
- login with admin
- click the add new order button, the New order dialog has opened correctly
- close the dialog and repeat the previous step

### Steps to reproduce

- git clone git@github.com:vaadin/bakery-app-starter-flow-spring.git
- mvn spring-boot:run -Pit,production
- login with admin
- click the add new order button, the New order dialog has opened correctly
- close the dialog and repeat the previous step, you can see an empty dialog 

### Environment

Vaadin version(s): V24.3.2 and V24.3.3

**Comment by web-padawan** [comment]
Could be a regression from https://github.com/vaadin/flow-components/pull/5635.

**Comment by ZheSun88** [comment]
yes, i just tested with 24.3.0.alpha2, the same behaivor. as that fix has been picked to 24.1.. so i guess the latest 24.1.14
![image](https://github.com/vaadin/flow-components/assets/31067185/b7326a63-66be-420f-9199-04c66db24abd)
